src/core/authentication/get.bearer.token.ts

```diff
@@ -0,0 +1,27 @@
+import { Configuration, PublicApi } from '@ory/kratos-client';
+
+export const getBearerToken = async (
+  kratosPublicUrl: string,
+  password_identifier: string,
+  password: string
+): Promise<string> | never => {
+  const kratos = new PublicApi(
+    new Configuration({
+      basePath: kratosPublicUrl,
+    })
+  );
+
```

Comment:
I'd move the checks for getting the bearer token from extendSession to here.
If a promise fails, I'd capture the error, and rethrow with a custom exception.